### DISCUSSION TITLE
Improvement of Git explanation

### DISCUSSION THREAD

**Discussion by seramm** [erlete/motorsport-programming-course#15] [Suggestions]
The Git explanation is reduced to what are the commits and the different sections inside a Git repository in GitHub. I believe that a small explanation of how Git states(working directory, staging area, local repo and remote repo) work would improve the understanding of Git and why people should commit before pushing, fetch changes, etc.
An image as the Figure 6 (the arrows are messed-up I don't know why but in the physical book are prettier) in Pro Git by Scott Chacon and Ben Straub would help understanding why _exempli gratia_ when using `git checkout` changes not commited will be lost or any of the situations explained before.

![image](https://user-images.githubusercontent.com/93453498/223989127-8f9839a7-9716-40dc-a900-5732cd3466ce.png)

I hope this helps improving the course!

**Comment by erlete** [comment]
Nice observation. Sure enough, the Git part of the course requires a bit more information. I'll look into it as soon as possible 👍🏽
